grpc/src/main/java/io/stargate/grpc/service/QueryHandler.java

```diff
@@ -192,4 +201,32 @@ private Parameters makeParameters(QueryParameters parameters, Optional<ClientInf
 
     return builder.tracingRequested(parameters.getTracing()).build();
   }
+
+  private CompletionStage<Void> waitForSchemaAgreement() {
+    CompletableFuture<Void> agreementFuture = new CompletableFuture<>();
+    waitForSchemaAgreement(Persistence.SCHEMA_AGREEMENT_WAIT_RETRIES, agreementFuture);
+    return agreementFuture;
+  }
+
+  private void waitForSchemaAgreement(
+      int remainingAttempts, CompletableFuture<Void> agreementFuture) {
+    if (persistence.isInSchemaAgreement()) {
+      agreementFuture.complete(null);
+      return;
+    }
+    if (remainingAttempts <= 1) {
+      agreementFuture.completeExceptionally(
+          Status.DEADLINE_EXCEEDED
+              .withDescription(
+                  "Failed to reach schema agreement after "
+                      + (200 * Persistence.SCHEMA_AGREEMENT_WAIT_RETRIES)
+                      + " milliseconds.")
+              .asException());
+      return;
+    }
+    executor.schedule(
+        () -> waitForSchemaAgreement(remainingAttempts - 1, agreementFuture),
+        200,
```

Comment:
shouldn't we make the interval time configurable? Or at least extract it to a constant and name it for better visibility of it?

Reply:
It's already configurable, `Persistence.SCHEMA_AGREEMENT_WAIT_RETRIES` can be overridden with a system property.